src/middleware.js

```diff
@@ -74,12 +74,20 @@ export function checkStatus(promise: Promise<Response>): Promise<Response> {
  * Takes a Promise which resolves to a Response and returns a Promise
  * which resolves to the JSON representation of the Response.
  * 
+ * If the response has a 204 No Content no response is parsed and instead
+ * a Promise which resolves to an empty object is returned.
+ * 
  * @param {Promise<Response>} Promise resolving to a Response
- * @return {Promise<JSON>} the JSON representation of the response body.
+ * @return {Promise<any>} the JSON representation of the response body.
  * @throws {Error} An Error indicating that the JSON could not be parsed. 
  */
-export function parseJSON(promise: Promise<Response>): Promise<JSON> {
+export function parseJSON(promise: Promise<Response>): Promise<any> {
   return promise.then((response: Response) => {
-    return response.json();
+
+    if (response.status === 204) {
```

Comment:
Maybe it is better to read the response header to check the content length.
This way you can be sure that there is an empty response or not.
Because I can send a 200 without any content as well for example.

Reply:
You are right you can send a 200 with any content but the 204 is more appropriate in that case. I've discussed this with @sptdevos and we concluded that when you actually know that the content is empty you should send a 204 from the back-end.